File: src/main/java/com/matchFit/post/dto/PostRequestDto.java
package com.matchFit.post.dto;

import com.matchFit.post.entity.Post;
import com.matchFit.post.entity.Sports;
import com.matchFit.post.entity.Status;
import com.matchFit.post.entity.Town;
import com.matchFit.user.entity.Gender;
import com.matchFit.user.entity.User;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PostRequestDto {
    private Long id;
    private String title;
    private String description;
    private String location;
    private String imageUrl;
    private Gender gender;
    private Sports sports;
    private Integer cost;
    private Status status;
    private Town town;
    private Integer maxPeople;
    private LocalDateTime date;

    public Post toEntity(User user) {
        Post post = new Post();
        post.setTitle(title == null ? "" : title);
        post.setDescription(description == null ? "" : description);
        post.setLocation(location == null ? "" : location);
        post.setImageUrl(imageUrl);
        if (gender == null) throw new IllegalArgumentException("gender is required");
        post.setGender(gender);
        if (sports == null) throw new IllegalArgumentException("sports is required");
        post.setSports(sports);
        post.setCost(cost == null ? 0 : cost);
        post.setStatus(status == null ? Status.OPEN : status);
        if (town == null) throw new IllegalArgumentException("town is required");
        post.setTown(town);
        post.setMaxPeople(maxPeople == null ? 0 : maxPeople);
        if (date == null) throw new IllegalArgumentException("date is required");
        post.setDate(date);
        post.setUser(user);
        return post;
    }
}

File: src/main/kotlin/com/matchFit/post/dto/PostRequestDto.kt
package com.matchFit.post.dto

import com.matchFit.post.entity.Post
import com.matchFit.post.entity.Sports
import com.matchFit.post.entity.Status
import com.matchFit.post.entity.Town
import com.matchFit.user.entity.Gender
import com.matchFit.user.entity.User
import java.time.LocalDateTime

class PostRequestDto {
    var id: Long? = null
    var title: String? = null
    var description: String? = null
    var location: String? = null
    var imageUrl: String? = null
    var gender: Gender? = null
    var sports: Sports? = null
    var cost: Int? = null
    var status: Status? = null
    var town: Town? = null
    var maxPeople: Int? = null
    var date: LocalDateTime? = null

    fun toEntity(user: User): Post {
        val post = Post()
        post.title = title ?: ""
        post.description = description ?: ""
        post.location = location ?: ""
        post.imageUrl = imageUrl
        post.gender = gender ?: throw IllegalArgumentException("gender is required")
        post.sports = sports ?: throw IllegalArgumentException("sports is required")
        post.cost = cost ?: 0
        post.status = status ?: Status.OPEN
        post.town = town ?: throw IllegalArgumentException("town is required")
        post.maxPeople = maxPeople ?: 0
        post.date = date ?: throw IllegalArgumentException("date is required")
        post.user = user
        return post
    }
}
